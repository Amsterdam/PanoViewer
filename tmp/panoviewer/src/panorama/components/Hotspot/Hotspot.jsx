import React from 'react';
import PropTypes from 'prop-types';
import './Hotspot.scss';

const Hotspot = ({ year, size, angle }) => (
  <div
    className="hotspot"
  >
    <button
      className={`c-hotspot c-hotspot--year-${year} qa-hotspot-button`}
    >
      <div
        className="qa-hotspot-rotation"
        style={{ width: `${size}px`, height: `${size}px`, transform: `rotateX(${angle}deg)` }}
      >
        <div
          className="c-hotspot__image"
          style={{ width: `${size}px`, height: `${size}px` }}
        />
      </div>

      <span className="u-sr-only">Navigeer naar deze locatie</span>
    </button>
  </div>
);

Hotspot.propTypes = {
  year: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  angle: PropTypes.number.isRequired
};

export default Hotspot;
