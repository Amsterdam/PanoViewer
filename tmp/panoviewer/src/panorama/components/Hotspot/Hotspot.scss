.c-hotspot {
  cursor: pointer;
  margin: 0 -20px;
  outline: none;
  padding: 0 20px;

  &:hover {
    animation: hotspot-animation 100ms linear;
    transform: scale(1.15);
  }
}

.c-hotspot--year-2016 {
  &:hover {
    .c-hotspot__image {
      background-image: url("assets/images/hotspots/2016-hover.svg");
    }
  }
}

.c-hotspot--year-2017 {
  &:hover {
    .c-hotspot__image {
      background-image: url("assets/images/hotspots/2017-hover.svg");
    }
  }
}

.c-hotspot--year-2018 {
  &:hover {
    .c-hotspot__image {
      background-image: url("assets/images/hotspots/2018-hover.svg");
    }
  }
}

.c-hotspot__image {
  background-size: cover;
  transform: scale(1);

  .c-hotspot--year-2016 & {
    background-image: url("assets/images/hotspots/2016.svg");
  }

  .c-hotspot--year-2017 & {
    background-image: url("assets/images/hotspots/2017.svg");
  }

  .c-hotspot--year-2018 & {
    background-image: url("assets/images/hotspots/2018.svg");
  }
}

@keyframes hotspot-animation {
  0% {
    transform: scale(1);
  }

  100% {
    transform: scale(1.15);
  }
}
