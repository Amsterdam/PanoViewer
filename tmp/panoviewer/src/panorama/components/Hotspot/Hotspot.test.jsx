import React from 'react';
import { shallow } from 'enzyme';
import Hotspot from './Hotspot';

describe('Hotspot', () => {
  it('should render', () => {
    const props = {
      year: '2020',
      size: 9,
      angle: 9
    };

    const component = shallow(
      <Hotspot {...props} />
    );
    expect(component).toMatchSnapshot();
  });
});
