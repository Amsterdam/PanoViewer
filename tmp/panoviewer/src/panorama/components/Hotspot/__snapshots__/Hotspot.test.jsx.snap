// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hotspot should render 1`] = `
<div
  className="hotspot"
>
  <button
    className="c-hotspot c-hotspot--year-2020 qa-hotspot-button"
  >
    <div
      className="qa-hotspot-rotation"
      style={
        Object {
          "height": "9px",
          "transform": "rotateX(9deg)",
          "width": "9px",
        }
      }
    >
      <div
        className="c-hotspot__image"
        style={
          Object {
            "height": "9px",
            "width": "9px",
          }
        }
      />
    </div>
    <span
      className="u-sr-only"
    >
      Navigeer naar deze locatie
    </span>
  </button>
</div>
`;
